framework/wazuh/core/indexer/agent.py

```diff
@@ -14,41 +17,44 @@ class AgentsIndex(BaseIndex):
     INDEX = 'agents'
     SECONDARY_INDEXES = []
 
-    async def create(self, id: str, key: str, name: str) -> Agent:
+    async def create(self, id: str, key: str, name: str, groups: str = None) -> dict:
         """Create a new agent.
 
         Parameters
         ----------
         id : str
-            Identifier of the new agent.
-        key : str
-            Key of the new agent.
+            New agent ID.
         name : str
-            Name of the new agent.
-
-        Returns
-        -------
-        Agent
-            The created agent instance.
+            New agent name.
+        key : str
+            New agent key.
+        groups : str
+            New agent groups.
 
         Raises
         ------
         WazuhError(1708)
             When already exists an agent with the provided id.
+
+        Returns
+        -------
+        agent_dict : dict
+            The created agent instance in a dictionary.
         """
-        agent = Agent(id=id, raw_key=key, name=name)
+        agent = Agent(raw_key=key, name=name, groups='default' + f',{groups}' if groups else '')
+        agent_dict = asdict(agent, dict_factory=remove_empty_values)
         try:
             await self._client.index(
                 index=self.INDEX,
-                id=agent.id,
-                body=asdict(agent),
+                id=id,
+                body=agent_dict,
                 op_type='create',
                 refresh='wait_for'
             )
         except exceptions.ConflictError:
             raise WazuhError(1708, extra_message=id)
-        else:
-            return agent
+        
+        return agent_dict
```

Comment:
Returning agent_dict because `dataclasses` are not JSON serializable

Reply:
Here we need to keep returning the Agent instance. In any case, it is possible to add a custom JSON decoder to serialize dataclass in the API responses.

Reply:
I added a condition to the `WazuhJSONEncoder` that converts dataclasses to dictionaries and modified this function. 

```
elif dataclasses.is_dataclass(obj):
            return dataclasses.asdict(obj)
```